packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -10,29 +10,56 @@
  * governing permissions and limitations under the License.
  */
 
-import {ToastStateBase} from '@react-types/toast';
-import {useState} from 'react';
+import {ReactNode, useRef, useState} from 'react';
+import {SpectrumToastProps, ToastState, ToastStateValue} from '@react-types/toast';
 
-
-interface ToastStateProps extends React.HTMLAttributes<HTMLElement> {
-  value?: ToastStateBase[]
+interface ToastStateProps {
+  value?: ToastStateValue[]
 }
 
-export function useToastState(props?: ToastStateProps) {
+const TOAST_TIMEOUT = 5000;
+
+export function useToastState(props?: ToastStateProps): ToastState {
   const [toasts, setToasts] = useState(props && props.value || []);
+  const toastsRef = useRef(toasts);
```

Comment:
you may have explained this at some other point, but why are we using a ref?
if it's because state changes after batches, you can use https://dmitripavlutin.com/react-usestate-hook-guide/#updating-the-state-with-a-callback
`setState(prevState => nextState);`

Reply:
Updating the state isn't the issue, it's an initial state within a closure issue. I might have tried that anyway.